src/components/UserForm/UserForm.tsx

```diff
@@ -0,0 +1,161 @@
+import { CompanyInterface, UserInterface, ValidationErrors } from '../../Interfaces/ObjectInterfaces';
+import { formValidation } from '../../services/formValidation';
+import { guIdGenerator } from '../../services/guidGenerator';
+
+export const UserForm: FC = () => {
+    let navigate = useNavigate();
+    let fetchedUserList = window.localStorage.getItem("storedUserList");
+    let fetchedCompanyList = window.localStorage.getItem("storedCompanyList");
+    let fetchedPositionList = window.localStorage.getItem("storedPositionList");
+
+    if (!fetchedUserList) {
+        fetchedUserList = '[]';
+    };
+    const parsedUserList: Array<UserInterface> = JSON.parse(fetchedUserList);
+
+    if (!fetchedCompanyList) {
+        fetchedCompanyList = '[]';
+    };
+    const parsedCompanyList: Array<CompanyInterface> = JSON.parse(fetchedCompanyList);
+
+    if (!fetchedPositionList) {
+        fetchedPositionList = '[]';
+    };
+    const parsedPositionList: string[] = JSON.parse(fetchedPositionList);
+
+    let { currentUserId } = useParams();
+
+    let newOrOldUser: UserInterface = {
+        id: '',
+        firstName: '',
+        lastName: '',
+        companyId: '',
+        companyName: '',
+        dOB: '1990-01-01',
+        position: '',
+        phoneNumber: '',
+    };
+
+    let currentUserFoundByID = parsedUserList.find((userFromList: UserInterface) => userFromList.id === currentUserId);
+
+    if (currentUserFoundByID) {
+        newOrOldUser = currentUserFoundByID
+    };
+
+    const [userInfo, setUserInfo] = useState<UserInterface>(newOrOldUser);
+
+    const [formErrors, setFormErrors] = useState<ValidationErrors>({ isValid: false });
+
+    const handleChangeSelect = (event: ChangeEvent<HTMLSelectElement>): void => {
+
+        if (event.target.name === 'companyName') {
+            setUserInfo({
+                ...userInfo,
+                [event.target.id]: event.target.value,
+                [event.target.name]: event.target.options[event.target.selectedIndex].text,
+            });
+        };
+
+        if (event.target.name !== 'companyName') {
+            setUserInfo({
+                ...userInfo,
+                [event.target.id]: event.target.value,
+            });
+        };
+    };
+
+    const handleChangeInput = (event: ChangeEvent<HTMLInputElement>): void => {
+
+        setUserInfo({
+            ...userInfo,
+            [event.target.id]: event.target.value,
+        });
+    }
+
+    const saveUser = () => {
+        setFormErrors(formValidation(userInfo));
+
+        if (formValidation(userInfo).isValid) {
+
+            if (currentUserId) {         // for editing old user
+                let userIndex = parsedUserList.findIndex((userFromList: UserInterface) => userFromList.id === currentUserId)
+                //updating company users array
+                let newCompanyList = parsedCompanyList.map((companyItem) => {
+                    if (companyItem.id === parsedUserList[userIndex].companyId) {
```

Comment:
You don't need to do everything if if else else etc
Here 2 sepeare if without else are quite fine

Reply:
ok, i see...